File: src/main/java/no/nav/arbeidsgiver/sykefravarsstatistikk/api/publiseringsdatoer/importering/PubliseringsdatoerImportService.java
package no.nav.arbeidsgiver.sykefravarsstatistikk.api.publiseringsdatoer.importering;

import lombok.extern.slf4j.Slf4j;
import no.nav.arbeidsgiver.sykefravarsstatistikk.api.importering.autoimport.DatavarehusRepository;
import no.nav.arbeidsgiver.sykefravarsstatistikk.api.publiseringsdatoer.PubliseringsdatoDbDto;
import no.nav.arbeidsgiver.sykefravarsstatistikk.api.publiseringsdatoer.PubliseringsdatoerRepository;
import org.springframework.stereotype.Component;

import java.util.List;

@Slf4j
@Component
public class PubliseringsdatoerImportService {

  private final PubliseringsdatoerRepository publiseringsdatoerRepository;
  private final DatavarehusRepository datavarehusRepository;

  public PubliseringsdatoerImportService(
      PubliseringsdatoerRepository publiseringsdatoerRepository,
      DatavarehusRepository datavarehusRepository) {
    this.publiseringsdatoerRepository = publiseringsdatoerRepository;
    this.datavarehusRepository = datavarehusRepository;
  }

  public void importerDatoerFraDatavarehus() {
    List<PubliseringsdatoDbDto> publiseringsdatoerFraDvh =
        datavarehusRepository.hentPubliseringsdatoerFraDvh();

    publiseringsdatoerRepository.oppdaterPubliseringsdatoer(publiseringsdatoerFraDvh);
  }
}

File: src/main/kotlin/no/nav/arbeidsgiver/sykefravarsstatistikk/api/applikasjon/publiseringsdatoer/PubliseringsdatoerImportService.kt
package no.nav.arbeidsgiver.sykefravarsstatistikk.api.applikasjon.publiseringsdatoer

import no.nav.arbeidsgiver.sykefravarsstatistikk.api.infrastruktur.database.PubliseringsdatoerRepository
import no.nav.arbeidsgiver.sykefravarsstatistikk.api.infrastruktur.datavarehus.DatavarehusPubliseringsdatoerRepository
import org.springframework.stereotype.Component

@Component
class PubliseringsdatoerImportService(
    private val publiseringsdatoerRepository: PubliseringsdatoerRepository,
    private val datavarehusPubliseringsdatoerRepository: DatavarehusPubliseringsdatoerRepository
) {
    fun importerDatoerFraDatavarehus() {
        val publiseringsdatoerFraDvh = datavarehusPubliseringsdatoerRepository.hentPubliseringsdatoerFraDvh()
        publiseringsdatoerRepository.overskrivPubliseringsdatoer(publiseringsdatoerFraDvh)
    }
}
